converters/toZigbee.js

```diff
@@ -2543,6 +2543,94 @@ const converters = {
+            case 'auto':
+                await sendTuyaCommand(entity, 257, 0, [1, 1/* on */]);
+                await utils.sleepMs(500);
+                await sendTuyaCommand(entity, 1028, 0, [1, 2/* auto */]);
+                break;
+            }
+        },
+    },
+    etop_thermostat_away: {
+        key: ['away'],
+        convertSet: async (entity, key, value, meta) => {
+            switch (value) {
+            case 'ON':
+                await sendTuyaCommand(entity, 257, 0, [1, 1/* on */]);
+                await utils.sleepMs(500);
+                await sendTuyaCommand(entity, 1028, 0, [1, 1/* away */]);
+                break;
+            case 'OFF':
+                await sendTuyaCommand(entity, 1028, 0, [1, 0/* manual */]);
+                break;
+            }
+        },
+    },
+    tuya_thermostat_weekly_schedule: {
+        key: ['weekly_schedule'],
+        convertSet: async (entity, key, value, meta) => {
+            const maxTransitions = utils.getMetaValue(entity, meta.mapped, 'weeklyScheduleMaxTransitions');
+            const supportedModes = utils.getMetaValue(entity, meta.mapped, 'weeklyScheduleSupportedModes');
+            const firstDayDpId = utils.getMetaValue(entity, meta.mapped, 'weeklyScheduleFirstDayDpId');
+
+            function transitionToData(transition) {
+                // Later it is possible to move converter to meta or to other place outside if other type of converter
+                // will be needed for other device. Currently this converter is based on ETOP HT-08 thermostat.
+                // see also fromZigbee.tuya_thermostat_weekly_schedule()
+                const minutesSinceMidnight = transition.transitionTime;
+                const heatSetpoint = Math.floor(transition.heatSetpoint * 10);
+                return [
+                    (minutesSinceMidnight & 0xff00) >> 8,
+                    minutesSinceMidnight & 0xff,
+                    (heatSetpoint & 0xff00) >> 8,
+                    heatSetpoint & 0xff,
+                ];
+            }
+            for (const [, daySchedule] of Object.entries(value)) {
+                const dayofweek = parseInt(daySchedule.dayofweek);
+                const numoftrans = parseInt(daySchedule.numoftrans);
+                let transitions = [...daySchedule.transitions];
+                const mode = parseInt(daySchedule.mode);
+                if (!supportedModes.includes(mode)) {
+                    throw new Error(`Invalid mode: ${mode} for device ${meta.options.friendlyName}`);
+                }
+                if (numoftrans != transitions.length) {
+                    meta.logger.warn(`Invalid numoftrans provided. Real: ${transitions.length} ` +
+                        `provided ${numoftrans} for device ${meta.options.friendlyName}`);
+                }
+                if (transitions.length > maxTransitions) {
+                    meta.logger.warn(`Reducing number of transitions from ${transitions.length} ` +
```

Comment:
Same as https://github.com/Koenkk/zigbee-herdsman-converters/pull/1607/files#r496827408

Reply:
What is behavior in standard cluster? 
Number of transistions is fixed and users should fill all of them, even they need few one?
For example. 
There is 4 transitions supported by the controller.
Is it possible to set only 2? In my case missing transitions will be filled by data from the last one.

Reply:
I expect that not all have to be filled, are you able to test this? (I don't have a Zigbee thermostat)

Reply:
I dont have Zigbee thermostat with pure clusters :(
In any case, I think it is possible to add similar behaviour `toZigbee.thermostat_weekly_schedule()` to prevent user-side errors.
How do you think?

Reply:
One problem, existing zigbee thermostats does not have `meta.thermostat.weeklyScheduleMaxTransitions`, but may be it is possible to take it from state...
Or may be be it's better to be more strict and throw error in such cases?

Reply:
I found only two of the supported thermostats which are able to buy.

- https://www.amazon.com/Centralite-3-Pearl-Touch-Thermostat/dp/B01G99KR94
- https://www.amazon.com/Zen-Within-Zen-01-W-Thermostat-Edition/dp/B00XCVV8CS

Reply:
- But for this thermostat, does it work when setting just 2 transitions? If yes, I think we can leave this out
- I don't see the added value for `meta.thermostat` yet https://github.com/Koenkk/zigbee-herdsman-converters/pull/1607/files#diff-6c9a6acf22f90d1c6e524d9f3c5c1745R1484 , this converter is not used by other devices and it's unclear when it can be if they would have different meta values. I think they can be embedded in the converter for now.

Reply:
* For this thermostat currently it will works with any number of transitions. If less then 4 transitions - it will pad missing with value with last one. If more than 4 - it will delete extra transitions.
* For other devices - I currently working on [HT-10](http://www.etopcontrols.com/thermostat/detail/HT-10.html), manufacturer told that it has same API, so this is reason I moved it to top level to the common converters. Also seems that `fromZigbee.tuyaThermostat` dp `112` and `113` (see image below) which are responsible for the schedule should be refactored to keep same API for the schedule. Problem is that they has only weekday/weekend (instead of full week) schedule. Also there is only one-way schedule in exisiting thermostats (fromZigbee). I will fix this if I will find any generic TuYa thermostat.

<img width="744" alt="image" src="https://user-images.githubusercontent.com/4232995/94611195-0d8d6c00-02aa-11eb-848f-681e063a6da7.png">

Reply:
But what happens if you just 2 transitions? 

>  If more than 4 - it will delete extra transitions.

I would throw an exception in this case

Reply:
This thermostat requres always 4 transitions.
So, I'm adding missing transition with same time and temperature:

**input:** 
```json
{
  "dayofweek": 1,
  "mode": 1,
  "numoftrans": 4,
  "transitions": [
    {
      "heatSetpoint": 11.5,
      "transitionTime": 0
    },
    {
      "heatSetpoint": 9,
      "transitionTime": 420
    },
  ]
}
```

**output:**
```json
{
  "dayofweek": 1,
  "mode": 1,
  "numoftrans": 4,
  "transitions": [
    {
      "heatSetpoint": 11.5,
      "transitionTime": 0
    },
    {
      "heatSetpoint": 9,
      "transitionTime": 420
    },
    {
      "heatSetpoint": 9,
      "transitionTime": 420
    },
    {
      "heatSetpoint": 9,
      "transitionTime": 420
    },
  ]
}
```

Reply:
If you still prefer to throw exception in both cases - I can do this.

Reply:
Only an exception when providng more than 4 transitions is enough. 

I have no more comments, can I mere this?